frmts/netcdf/netcdfdataset.cpp

```diff
@@ -9104,7 +9104,7 @@ netCDFDataset *netCDFDataset::CreateLL(const char *pszFilename, int nXSize,
 
     CPLReleaseMutex(hNCMutex);  // Release mutex otherwise we'll deadlock with
         // GDALDataset own mutex.
-    netCDFDataset *poDS = new netCDFDataset();
+    auto poDS = std::make_unique<netCDFDataset>();
```

Comment:
I'm not totally sure this change is safe. See the comments around the explicit "delete poDS" about mutex deadlocks

Reply:
```
// Release mutex otherwise we'll deadlock with
        // GDALDataset own mutex.
```
Is this referring to use of `hNCMutex` by `netCDFDataset` ? Or is it referring to the `m_poPrivate->hMutex` in `GDALDataset` ?

Reply:
I believe this was a deadlock between hNCMutex and https://github.com/OSGeo/gdal/blob/732d82cf7486be5ee12e1d67c60f09f051072e0e/gcore/gdaldataset.cpp#L324

Reply:
I've removed this change.